src/actions/bitcoinActions.js

```diff
@@ -0,0 +1,169 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import Toast from 'components/Toast';
+import {
+  UPDATE_BITCOIN_BALANCE,
+  REFRESH_THRESHOLD,
+  SET_BITCOIN_ADDRESSES,
+} from 'constants/bitcoinConstants';
+import {
+  keyPairAddress,
+  getAddressUtxos,
+  importKeyPair,
+  exportKeyPair,
+  rootFromMnemonic,
+  transactionFromPlan,
+  sendRawTransaction,
+} from 'services/bitcoin';
+import Storage from 'services/storage';
+
+import type { Dispatch, GetState } from 'reducers/rootReducer';
+import type {
+  BitcoinReducerAction,
+  SetBitcoinAddressesAction,
+  UpdateBitcoinBalanceAction,
+} from 'reducers/bitcoinReducer';
+import type { Wallet } from 'models/Wallet';
+import type { BitcoinTransactionPlan, BitcoinUtxo } from 'models/Bitcoin';
+
+import { saveDbAction } from 'actions/dbActions';
+
+const storage = Storage.getInstance('db');
+
+type BitcoinStore = {
+  keys?: { [key: string]: string },
+};
+
+const saveDb = (data: BitcoinStore) => {
+  return saveDbAction('bitcoin', data, true); // TODO: +true+ required?
+};
+
+const loadDb = async (): Promise<BitcoinStore> => {
+  return storage.get('bitcoin');
+};
+
+const setBitcoinAddresses = (addresses: string[]): SetBitcoinAddressesAction => ({
+  type: SET_BITCOIN_ADDRESSES,
+  addresses,
+});
+
+const updateBitcoinBalance = (
+  address: string,
+  unspentTransactions: BitcoinUtxo[],
+): UpdateBitcoinBalanceAction => ({
+  type: UPDATE_BITCOIN_BALANCE,
+  address,
+  unspentTransactions,
+});
+
+export const initializeBitcoinWalletAction = (wallet: Wallet) => {
```

Comment:
what type of wallet is it? Generic, ethereum, bitcoin?

Reply:
I mean in params

Reply:
Good catch, this is actually used from the views so there's not much context.

I might be confusing things here, let's see if this makes sense. We need to make sure the BTC addresses are generated in an idempotent way. My understanding is that we can use the same mnemonic we use for generating the Ethereum address, since it's the one that's never gonna change.

The [checkAuthAction](https://github.com/pillarwallet/pillarwallet/blob/feature/btc/src/screens/Bitcoin/Bitcoin.js#L96) passes this param when validating the pin, which I'm not sure what it is but the `Wallet` type seems to work fine.

We'd use this action in a way similar to what we had on the MVP: https://github.com/pillarwallet/pillarwallet/blob/feature/btc/src/screens/Bitcoin/Bitcoin.js#L96 (You can find some extra context there, althought that's not *exactly* how's gonna be the new implementation)

Reply:
Btw, if the wallet was imported from the private key - it won't have a mnemonic

Reply:
Damn, you're right 🤔. So we're gonna need to have independent import/export for BTC and we could still store that into `Wallet` with a `btc` prefix, for example (i.e. `btcMnemonic`, `btcPrivateKey`) or, even under a new key (i.e. `{ "btc": { "mnemonic": "...", "privateKey": ... } }`).

What do you think?

Reply:
You could just rename `Wallet` into `GenericWallet` or `EthereumWallet` that has a `mnemonic` property marked as optional

Reply:
Alright, updated.